### PR TITLE
[TYR]: TyrUserEvent and RabbitMqHandler implemented (and refactoring)

### PR DESCRIPTION
**Pull request by lrocheWB**
# Introduction
- TyrUserEvent and RabbitMqHandler implemented: able to publish events through RabbitMQ. This will synchronize FenrirApi which manage users (https://github.com/CanalTP/FenrirApi). For now, only the methods "put", "post" and "delete" are considered.
- Refactoring "source/tyr/tyr/binarisation.py" file

Related to:
https://github.com/CanalTP/FenrirApi/pull/14

screenshot from 2016-05-18 18 25 48

cc @Netmisa ;)

### CODE REVIEW

**Comment by antoine-de** [comment]
Jenkins, test this please

File: source/tyr/tyr/rabbit_mq_handler.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+from kombu import Exchange, Connection, Producer
+import logging
+
+from flask import current_app
+
+class RabbitMqHandler(object):
+
+    def __init__(self, exchange_name, type='direct', durable=True):
+        try:
+            self._connection = Connection(current_app.config['CELERY_BROKER_URL'])
```
Comment (kinnou02) [inline source/tyr/tyr/rabbit_mq_handler.py]: You should take the connection string in parameters.

File: source/tyr/tyr/rabbit_mq_handler.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            self._producer = Producer(self._connection)
+            self._task_exchange = Exchange(name=exchange_name, type=type, durable=durable)
+        except Exception:
+            logging.exception('Unable to activate the producer')
```
Comment (kinnou02) [inline source/tyr/tyr/rabbit_mq_handler.py]: You always should create a logger.

File: source/tyr/tyr/rabbit_mq_handler.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    def publish(self, payload, routing_key=None, serializer='json'):
+        publish = self._connection.ensure(self._producer, self._producer.publish, errback = self.errback, max_retries=3)
+        publish(payload,
+                serializer=serializer,
+                exchange=self._task_exchange,
+                declare=[self._task_exchange],
+                routing_key=routing_key)
+
+        logging.info("Sent: %s" % payload)
```
Comment (kinnou02) [inline source/tyr/tyr/rabbit_mq_handler.py]: It's more a debug log.

File: source/tyr/tyr/rabbit_mq_handler.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+            self._task_exchange = Exchange(name=exchange_name, type=type, durable=durable)
+        except Exception:
+            logging.exception('Unable to activate the producer')
+            raise
+
+    def errback(exc, interval):
+        logging.info('Error: %r', exc, exc_info=1)
+        logging.info('Retry in %s seconds.', interval)
+
+    def publish(self, payload, routing_key=None, serializer='json'):
```
Comment (kinnou02) [inline source/tyr/tyr/rabbit_mq_handler.py]: The default value for serializer seem to bo `None`: http://docs.celeryproject.org/projects/kombu/en/latest/reference/kombu.html#kombu.Producer.publish

Comment (lrocheWB) [inline source/tyr/tyr/rabbit_mq_handler.py]: I confused with: http://docs.celeryproject.org/projects/kombu/en/latest/reference/kombu.html#message-producer
Thx

File: source/tyr/tyr/rabbit_mq_handler.py (hunk 1 above)
Comment (lrocheWB) [inline source/tyr/tyr/rabbit_mq_handler.py]: Done.

File: source/tyr/tyr/rabbit_mq_handler.py (hunk 2 above)
Comment (lrocheWB) [inline source/tyr/tyr/rabbit_mq_handler.py]: Done.

File: source/tyr/tyr/rabbit_mq_handler.py (hunk 3 above)
Comment (lrocheWB) [inline source/tyr/tyr/rabbit_mq_handler.py]: Remove.

**Pull request merged**